src/openrct2/game.c

```diff
@@ -1357,8 +1359,8 @@ bool game_load_save_or_scenario(const utf8 * path)
 
 static void game_load_or_quit_no_save_prompt_callback(sint32 result, const utf8 * path)
 {
-    if (result == MODAL_RESULT_OK && game_load_save_or_scenario(path)) {
-        gFirstTimeSave = 0;
+    if (result == MODAL_RESULT_OK) {
+        game_load_save_or_scenario(path);
```

Comment:
I see. game_load_save_or_scenario() calls either game_load_save() scenario_load_and_play_from_path(), both of which indeed set gFirstTimeSave.